kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/ssl/SslManagerServiceImpl.java

```diff
@@ -629,27 +629,20 @@ private boolean changeKeyStorePassword() {
             }
         }
 
-        if (oldPassword == null) {
-            logger.warn("null old password");
-            return false;
-        }
-
-        if (!Arrays.equals(oldPassword, newPassword)) {
+        if (oldPassword != null && !Arrays.equals(oldPassword, newPassword)) {
```

Comment:
Can `newPassword` be null at this point? Should you check?

Reply:
new password is initialised as old password

Reply:
and changed later on. It's unlikely to be null but just in case...